QUARKUS-4592.md

```diff
@@ -0,0 +1,112 @@
+Main focus is on testing features documented in the TLS registry reference. That is:
+- configure HTTP server to use HTTPS protocol
+- configure management interface to use HTTPS protocol
+- configure mTLS
+- configure client to use HTTPS protocol; here we should concentrate on:
+  - gRPC
+  - Quarkus REST client
+- accessing TLS registry API from CDI in application code (both named and the default one)
+- configure keystore, truststore, cipher suits and protocol
+- use PEM / PKCS12 / JKS keystore and truststore
+- test certificate reloading in DEV mode
+- use OpenShift serving certificates to generate and renew TLS certificates automatically
+- generate certificates and DEV CA with Quarkus CLI, then run app and make sure the certs are configured and valid
+- support for the TLS registry to the mailer extension
+
+Additionally, secured communication between Quarkus application and a Strimzi server must be tested.
+We want to test registry where we already test extension-specific TLS configuration, while test for every single supported extension may not be possible due to time limitations.
+For example, secured communication with AMQ broker, REDIS cache and OTLP collector will be tested if time allows, or delayed (with opened Quarkus QE Test Suite issue).
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Focus on exploratory testing of the feature
+- Ensure good user experience and simplicity of use
+
+## Existing test coverage
+Quarkus unit test coverage is sufficient. As much as I am thinking how much do we need to test.
+Integration tests are spare and sometimes missing, for example I didn't find Quarkus REST client or OTLP integration 
+tests (please don't confuse with unit tests placed in IT modules).
+Quarkus QE Test Suite sufficiently covers TLS configuration for HTTP server, management interface, Kafka.
+Smoke test TLS coverage for REST clients and gRPC is also present.
+
+### Impact on test suites and testing automation
+I am very reluctant to mirror all TLS tests for both extension-specific and TLS registry.
+My plan is to mix it all, where there is duplicity, try a bit from both, when there is just one specific test, add new tests.
+Several new test classes will be added starting integrations (e.g. Kafka), DEV mode tests that require reload and waiting period (before new certs are detected).
+New OpenShift tests are required to test integration with serving certificates.
+
+### Impact on resources
+Counting in OpenShift, native, JVM and DEV mode, less than half of an hour will be added to the test execution.
+
+## Future considerations
+- [OIDC integration](https://github.com/quarkusio/quarkus/issues/41001) is yet to be implemented
+- [Reactive JDBC clients integration for DB2 and MSSQL extensions](https://github.com/quarkusio/quarkus/issues/41003)
+- Let’s Encrypt support (I don't want our tests to call external authority for now due to network connectivity demands and stability of tests)
```

Comment:
+1, would be still good to try it locally when exploring the TLS feature.

Reply:
added it into the TP